ddtrace/tracer/span.go

```diff
@@ -427,8 +429,12 @@ func (s *Span) setTagError(value interface{}, cfg errorConfig) {
 		setError(true)
 		s.setMeta(ext.ErrorMsg, v.Error())
 		s.setMeta(ext.ErrorType, reflect.TypeOf(v).String())
+		// if !cfg.noDebugStack {
+		// 	s.setMeta(ext.ErrorStack, takeStacktrace(cfg.stackFrames, cfg.stackSkip))
+		// }
+		tracerErr := errortrace.Wrap(v, cfg.stackFrames, cfg.stackSkip)
 		if !cfg.noDebugStack {
-			s.setMeta(ext.ErrorStack, takeStacktrace(cfg.stackFrames, cfg.stackSkip))
+			s.setMeta(ext.ErrorStack, tracerErr.Stack().String())
 		}
 		switch v.(type) {
 		case xerrors.Formatter:
```

Comment:
let's add another case here to check for our new error type to add metadata as we expect